pytest_reportportal/service.py

```diff
@@ -231,6 +266,79 @@ def _stop_if_necessary(self):
         except queue.Empty:
             pass
 
+
+    @staticmethod
+    def _add_item_hier_parts_dirs(item, hier_flag, dirs_level, report_parts, dirs_parts, rp_name=""):
+
+        parts_dirs = PyTestServiceClass._get_item_dirs(item)
+        dir_path = item.fspath.new(dirname="", basename="", drive="")
+        for dir_name in parts_dirs[dirs_level:]:
+            dir_path = dir_path.join(dir_name)
+            path = str(dir_path)
+
+            if hier_flag:
+                if path in dirs_parts:
+                    item_dir = dirs_parts[path]
+                else:
+                    item_dir = File(dir_name, nodeid=dir_name, session=item.session, config=item.session.config)
+                    item_dir._rp_name = dir_name
+                    dirs_parts[path] = item_dir
+
+                report_parts.append(item_dir)
+            else:
+                rp_name = path
+
+        return rp_name
+
+    @staticmethod
+    def _add_item_hier_parts_parametrize(item, report_parts, tests_parts, rp_name=""):
+
+        for mark in item.own_markers:
+            if mark.name == 'parametrize':
+                ch_index = item.nodeid.find("[")
+                test_fullname = item.nodeid[:ch_index if ch_index > 0 else len(item.nodeid)]
+                test_name = item.originalname
+
+                rp_name += ("::" if rp_name else "") + test_name
+
+                if test_fullname in tests_parts:
```

Comment:
is tests_parts variable only for caching?

Reply:
See next comment. Similar. It allow us to do not create new test hierarchy item each time when we see test item that we already saw for same test but with another parameters.
Example:
dir1/dir2/module1.py::TestClass1::test1 [param1=1]
dir1/dir2/module1.py::TestClass1::test1 [param1=2]
After 1) in `tests_parts` we will have `{'dir1/dir2/module1.py::TestClass1::test1': ..., }`. And we will use in 2)